installer/Makefile

```diff
@@ -0,0 +1,119 @@
+GOLINT ?= $(GOPATH)/bin/golint$(BIN_ARCH)
+
+.PHONY: gas ova
+
+LDFLAGS := $(shell BUILD_NUMBER=${BUILD_NUMBER} TAG=${TAG} $(BASE_DIR)/scripts/version-linker-flags.sh)
+
+ovfenv := $(BIN)/ovfenv
+ova-ui := $(BIN)/ova-ui
+toolbox := $(BIN)/toolbox
+rpctool := $(BIN)/rpctool
+gas: $(GAS)
+ovfenv: $(ovfenv)
+ova-ui: $(ova-ui)
+toolbox: $(toolbox)
+rpctool: $(rpctool)
+tools: $(DEP) $(GOLINT) $(GAS)
+all: golint gofmt govet gas $(ovfenv) $(ova-ui) $(toolbox) $(rpctool)
+
+vendor: $(DEP)
+	@echo restoring vendor
+	$(DEP) ensure
+
+gas: $(GAS)
+	@echo running go AST tool
+	@$(GAS) -quiet lib/... ovatools/... pkg/... toolbox/... 2> /dev/null
+
+golint: $(GOLINT)
+	@echo checking go lint...
+	@$(call golintf,sigs.k8s.io/cluster-api-provider-vsphere/installer/lib/...)
+	@$(call golintf,sigs.k8s.io/cluster-api-provider-vsphere/installer/ovatools/...)
+	@$(call golintf,sigs.k8s.io/cluster-api-provider-vsphere/installer/pkg/...)
+	@$(call golintf,sigs.k8s.io/cluster-api-provider-vsphere/installer/toolbox/...)
+
+gofmt:
+	@echo checking gofmt...
+	@! gofmt -d -e -s $$(find . -mindepth 1 -maxdepth 1 -type d -not -name vendor) 2>&1 | egrep -v '^$$'
+
+govet:
+	@echo checking go vet...
+	@$(GO) tool vet -all -lostcancel -tests $$(find . -mindepth 1 -maxdepth 1 -type d -not -name vendor)
+
+# Generate Go package dependency set, skipping if the only targets specified are clean and/or distclean
+# Caches dependencies to speed repeated calls
+define godeps
+	$(call assert,$(call gmsl_compatible,1 1 7), Wrong GMSL version) \
+	$(if $(filter-out clean distclean mrrobot mark sincemark .DEFAULT,$(MAKECMDGOALS)), \
+		$(if $(call defined,dep_cache,$(dir $1)),,$(info Generating dependency set for $(dir $1))) \
+		$(or \
+			$(if $(call defined,dep_cache,$(dir $1)), $(info Using cached Go dependencies) $(wildcard $1) $(call get,dep_cache,$(dir $1))),
+			$(call set,dep_cache,$(dir $1),$(shell $(BASE_DIR)scripts/go-deps.sh $(dir $1) $(MAKEFLAGS))),
+			$(info Cached Go dependency for $(dir $1): $(call get,dep_cache,$(dir $1))),
+			$(wildcard $1) $(call get,dep_cache,$(dir $1))
+		) \
+	)
+endef
+
+clean:
+	@echo removing binaries
+	rm -rf $(BIN)
+
+# exit 1 if golint complains about anything other than comments
+golintf = $(GOLINT) $(1) | sh -c "! grep -v 'lib/apiservers/portlayer/restapi/operations'" | sh -c "! grep -v 'lib/config/dynamic/admiral/client'" | sh -c "! grep -v 'should have comment'" | sh -c "! grep -v 'comment on exported'" | sh -c "! grep -v 'by other packages, and that stutters'" | sh -c "! grep -v 'error strings should not be capitalized'"
+
+$(ovfenv): $(call godeps,ovatools/ovfenv/*.go)
+	@echo building ovfenv linux...
+	@GOARCH=amd64 GOOS=linux $(TIME) $(GO) build $(RACE) -ldflags "$(LDFLAGS)" -o ./$@ ./$(dir $<)
+
+$(ova-ui): $(call godeps,ovatools/ova-ui/*.go)
+	@echo building ova-ui
+	@GOARCH=amd64 GOOS=linux $(TIME) $(GO) build $(RACE) -ldflags "$(LDFLAGS)" -o ./$@ ./$(dir $<)
+
+$(toolbox): $(call godeps,toolbox/*.go)
```

Comment:
I'm a fan of `toolbox` for simple use cases like this one, it also gives us the ability to extend its functionality (like [this](
https://github.com/vmware/govmomi/tree/master/toolbox#httproundtripper)) to accommodate more complex automation with Cluster API

Reply:
Hi @frapposelli / @zjs,

Reply:
Hi @frapposelli / @zjs,

FWIW, yakity uses `open-vm-tools`, and I often have issues with them taking a while to come online in terms of visibility to the vCenter. That said, I use `open-vm-tools` because it's one less thing...

Reply:
Note that toolbox does support all of the guest operations.  It is possible to only expose a subset (as VIC does) or disable all (as this PR does).  The main issue is that toolbox only has a hook for in-guest authentication, but leaves it to the consumer to implement. 
https://github.com/vmware/govmomi/tree/master/toolbox#guestfilemanager-object
https://github.com/vmware/govmomi/tree/master/toolbox#guestprocessmanager

Regarding the visibility in vCenter, open-vm-tools takes ~30 sec (last I checked) to initially report IP addresses, toolbox is around 1 sec or less.